### DISCUSSION TITLE
How to call py script from Nextjs using exec?

### DISCUSSION THREAD

**Discussion by krishnaacharyaa** [vercel/next.js#61749] [Help]
### Summary

I am trying to run simple py/sh scripts from nextjs using exec but i am not able to achieve it.

Kind help would be much appreciated.

Process is simple: 
>I am calling server Actions in my code, which executes the script

page.tsx
```
"use client";

import { pythonExec } from "./action";

export default function PythonClicker() {
  return <div onClick={() => pythonExec()}>Hello there</div>;
}

```

server action
```
"use server";

// import { exec } from "child_process";
export const pythonExec = async () => {
  const { exec } = require("child_process");
  console.log("I am called  ");
  exec('"sample-script.py"');
};
```

sample-script.py
```
print("Hello world")

```

p.s: I am called is being logged, only the "Hello world" is not getting logged.
I also want to return values from the script and use consume it in my server action 

### Example

https://github.com/krishnaacharyaa/python-nextjs

**Comment by icyJoseph** [comment]
Hi,

You have to do it like this:

```js
const { exec } = require("child_process"); // or using import { exec } from "child_process";

const child = exec("python3 example.py");

child.stdout.pipe(process.stdout);
```

The problem you have is that you are not passing the script into the python interpreter, you are just using the python script name.

Furthermore, you need to get a hold of the stdout, and stderr, of the child process running the python script, and use it back on your current process.
